### PR TITLE
raft: Fix race in raftclusterer tests

### PR DESCRIPTION
**Pull request by babbageclunk**
## Description of change

The race was caused by resetting s.reqs to a new channel in the test
setup when a subscribe callback had previously read s.reqs from a
different goroutine (in a previous test). Avoid it by having the
callback send to a local variable so that setting s.reqs in a subsequent
call to SetUpTest doesn't overwrite the old channel.

## QA steps

Ran under race checker before and after fix.

### CODE REVIEW

File: worker/raft/raftclusterer/worker_test.go [hunk 1]
```diff
@@ -84,15 +84,21 @@ var _ = gc.Suite(&WorkerSuite{})
...
+	unsubscribe, err := s.hub.Subscribe(
 		apiserver.DetailsRequestTopic,
 		func(topic string, req apiserver.DetailsRequest, err error) {
 			c.Check(err, jc.ErrorIsNil)
-			s.reqs <- req
+			reqs <- req
 		},
 	)
+	c.Assert(err, jc.ErrorIsNil)
+	s.AddCleanup(func(c *gc.C) { unsubscribe() })
```
Comment (howbazaar) [inline worker/raft/raftclusterer/worker_test.go]: The unsubscribe should be sufficient, as once unsubscribe occurs, any pending calls that haven't been processed get cleared out. So, once unsubscribe returns there will be no more calls to the callback. You don't need the local variable.

Comment (babbageclunk) [reply worker/raft/raftclusterer/worker_test.go]: Nope - that was what I thought initially too but it still failed. I think the problem the race checker is pointing out is:
* goroutine A (the main one) writes s.reqs (setting it to a new channel).
* goroutine B (inside the hub) reads s.reqs (to send on it)
* goroutine A writes s.reqs again in the setup of the next test.
* goroutine B reads it again in the callback from publishes in the tests.

So even though we've unsubscribed, there's no actual synchronisation around the reads and writes of s.reqs.

Comment (babbageclunk) [reply worker/raft/raftclusterer/worker_test.go]: (Demonstrated to @howbazaar to convince him.)

**Comment by babbageclunk** [comment]
$$merge$$

**Pull request merged**